packages/slate-react/src/components/editable.tsx

```diff
@@ -180,21 +179,23 @@ export const Editable = (props: EditableProps) => {
     // Otherwise the DOM selection is out of sync, so update it.
     const el = ReactEditor.toDOMNode(editor, editor)
     state.isUpdatingSelection = true
-    domSelection.removeAllRanges()
 
     const newDomRange = selection && ReactEditor.toDOMRange(editor, selection)
 
     if (newDomRange) {
-      if (Range.isBackward(selection!)) {
-        setReverseDomSelection(newDomRange, domSelection)
-      } else {
-        domSelection.addRange(newDomRange!)
-      }
+      domSelection.setBaseAndExtent(
+        newDomRange.startContainer,
+        newDomRange.startOffset,
+        newDomRange.endContainer,
+        newDomRange.endOffset
+      )
```

Comment:
IIUC, although `setBaseAndExtent` can create a reversed selection, the start and end of the range (`newDomRange`) will always be in the forwards direction so this will always be creating a forwards selection.

Previously, `Range.isBackward` was used to check the orientation of the selections's anchor and focus (note the difference from a range's start and end). If the selections is backwards, the end of the range needs to be used as the anchor rather than the start.

Reply:
I've updated the code to correct for that! Nice catch 👍

That behavior defied my intuition... If the `setBaseAndExtent` is capable of keeping track of the focus and anchor, it stands to reason (and wrongly so) that it was supported by `DOMRange`. It seems like that's a limitation of HTML itself, not just `newDomRange`.